pyccel/ast/numpyext.py

```diff
@@ -367,7 +367,7 @@ def __init__(self, arg, dtype=None, order='C'):
 
         # Verify dtype and get precision
         if dtype is None:
-            dtype = arg.dtype
+            dtype = DtypePrecisionToCastFunction[str(arg.dtype).capitalize()][arg.precision]
```

Comment:
In other places where this dictionary is accessed the key is: `arg.dtype.name` are you sure this will always give the same result as `str(arg.dtype).capitalize()`?

Reply:
I didn't notice that attribute, changing `str(arg.dtype).capitalize()` for `arg.dtype.name` seems like the best thing to do.